src/common/campaign/common.ts

```diff
@@ -131,3 +137,33 @@ export const getReferenceTags = (
       throw new ValidationError('Unknown campaign type to estimate reach');
   }
 };
+
+export interface UserCampaignData {
+  impressions: number;
+  clicks: number;
+  spend: number;
+  users: number;
+}
+
+export const getUserCampaignData = async (
+  ctx: AuthContext,
+): Promise<UserCampaignData> => {
+  const result = await queryReadReplica(ctx.con, ({ queryRunner }) =>
```

Comment:
You ok with this coming from read replica? I'm not sure how often it processes this/how realtime it needs to be?